packages/config/src/layer2s/common/ProjectDiscovery.ts

```diff
@@ -0,0 +1,57 @@
+import { ContractParameters, ProjectParameters } from '@l2beat/types'
+import fs from 'fs'
+import path from 'path'
+export class ProjectDiscovery {
+  private readonly discovery: ProjectParameters
+  constructor(project: string) {
+    this.discovery = this.getDiscoveryJson(project)
+  }
+
+  private getDiscoveryJson(project: string): ProjectParameters {
+    const discoveryFile = fs.readFileSync(
+      path.resolve(`../backend/discovery/${project}/discovered.json`),
+      'utf-8',
+    )
+
+    return JSON.parse(discoveryFile) as ProjectParameters
+  }
+
+  getContractByName(name: string): ContractParameters {
```

Comment:
That is something I wanted to discuss, I was thinking about it but on the other hand, the address can change and the name will prevail the same so it will be more dynamic